.github/workflows/macos.yml

```diff
@@ -71,9 +84,37 @@ jobs:
 
       - run: make
 
+      # Launchable CLI requires Python and Java
+      # https://www.launchableinc.com/docs/resources/cli-reference/
+      - name: Set up Python
+        uses: actions/setup-python@871daa956ca9ea99f3c3e30acb424b7960676734 # v5.0.0
+        with:
+          python-version: "3.10"
+      - name: Set up Java
+        uses: actions/setup-java@7a445ee88d4e23b52c33fdc7601e40278616c7f8 # v4.0.0
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+      - name: Set up Launchable
+        id: setup_launchable
+        run: |
+          set -ex
+          pip install launchable
+          launchable verify
+          launchable record build --name ${GITHUB_PR_HEAD_SHA}
+          : # To avoid file name conflicts, we use a random number here as a ID.
+          random_number=$RANDOM
+          echo "random_number=$random_number" >> $GITHUB_OUTPUT
+          echo "TEST_WITH_LAUNCHABLE=--launchable-test-reports=$random_number.json" >> "$GITHUB_ENV"
```

Comment:
The reason of using bash's `$RANDOM` variable is that Lunchable wants to make the ID of the json file name unique permanently? Or to make the ID unique only between workflows (ID: run ID?) in the same build? I checked this PR's 2 workflows below in the same build. I think it is technically possible for  multiple workflows use the same the file name.

make (check, macos-12): https://github.com/ruby/ruby/actions/runs/7927218713/job/21643235510?pr=9777#step:10:254

```
+ echo TEST_WITH_LAUNCHABLE=--launchable-test-reports=22433.json
```

make (check, macos-13): https://github.com/ruby/ruby/actions/runs/7927218713/job/21643236338?pr=9777#step:10:254

```
+ echo TEST_WITH_LAUNCHABLE=--launchable-test-reports=21795.json
```

So, if we guarantee the unique ID technically between the workflows or permanently, we may use the workflow's ID environment variable `GITHUB_RUN_ID` as a part of the json file name?

https://docs.github.com/en/actions/learn-github-actions/variables
> GITHUB_RUN_ID
> A unique number for each workflow run within a repository. This number does not change if you re-run the workflow run. For example, 1658821493.

```
echo TEST_WITH_LAUNCHABLE=--launchable-test-reports=${GITHUB_RUN_ID}.json
```

```
echo TEST_WITH_LAUNCHABLE=--launchable-test-reports=${GITHUB_RUN_ID}_${RANDOM}.json
```

Reply:
It does completely make sense. Updated.

Reply:
Regarding the uniqueness of the report files needed for Launchable integration, I initially considered incorporating randomness to prevent potential conflicts between different jobs, such as 'make test all' on different macOS versions. However, according to GitHub's documentation on storing workflow data as artifacts:

> Artifacts allow you to persist data after a job has completed, and share that data with another job in the same workflow. 

(Source: [GitHub Docs](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts))

This implies that files are not shared between different jobs unless utilizing the artifact library.

Consequently, the concern about file overriding between jobs becomes unnecessary. It seems we can safely disregard this aspect moving forward.

Reply:
Ah, okay. Thanks for explaining it. So, your concern was a possibility to share a file and override it between jobs on the GitHub Actions' hosts. I was assuming your concern was a possibility to upload the reports file from each host on GitHub Actions into your Lunchable server, then to override a file there.

So, if there is no concern to override the file between jobs, what do you think about changing the curent reports file name `${GITHUB_RUN_ID}.json` to the fixed name such as `launchable-reports.json`? I just would like to add the prefix "launchable-" to avoid a possible file conflict from files generated by other programs in the ruby/ruby.

Reply:
Updated

Reply:
Okay I confirmed the reports file name was the `launchable_reports.json` at the latest commit on this PR.